hydromt/raster.py

```diff
@@ -2247,6 +2248,245 @@ def tile_window(shape, px):
         with open(join(root, f"{mName}.yml"), "w") as f:
             yaml.dump({mName: yml}, f, default_flow_style=False, sort_keys=False)
 
+    def to_osm_tiles(
+        self,
+        root: str,
+        zoom_levels: list,
+        bbox: tuple = (),
+        method: str = "nearest",
+        min_lvl: int = 5,
+        driver="GTiff",
+        **kwargs,
+    ):
+        """Export rasterdataset to tiles in an osm structure.
+
+        Parameters
+        ----------
+        root : str
+            Path where the database will be saved
+            Database yml will be put one directory above
+        zoom_levels : list
+            Zoom levels to be put in the database
+        bbox : tuple, optional
+            A region where the tiles should be generated for
+            (if absent, the entire input raster dataset will be tiled)
+        method : str, optional
+            How to resample the data when downscaling.
+            E.g. 'nearest' for resampling with the nearest value
+        min_lvl : int, optional
+            Minimal starting level from to where to downscale from.
+            Increase if memory issues are encountered.
+        driver : str, optional
+            GDAL driver (e.g., 'GTiff' for geotif files), or 'netcdf4' for netcdf files.
+        **kwargs
+            Key-word arguments to write raster files
+        """
+        resampling = getattr(Resampling, method, None)
+        if resampling is None:
+            raise ValueError(f"Resampling method unknown: {method}.")
+        px_size = 256  # Fixed resolution for osm
+
+        assert self._obj.ndim == 2, "Only 2d datasets are accepted..."
+        obj = self._obj.copy()
```

Comment:
Not quite sure if this is true, but it doesn't seem like you actually have to mutate `obj` in any way in this function. If that is the case and you are having memory issues, you could see if you can avoid making a full copy, and just use a reference to the object directly

Reply:
General remark from the user point of view; the advantage of using the OSM-tiles (basically just a specific format of xyz-tiles) is that the tiles are fixed in position and size (256x256), and ofcourse coordinate reference system. I think that allows for having different formats, like the .png format, instead of the netcdf4 or geotiffs, that do NOT include any spatial information. These .pngs can then very easily be visualized in webviewers or any other mapbox/leaflet like applications.

I'm not sure if the method here, using the VRT file, still works for binary formats or the .png format that doesnt have any spatial information included? Webviewers but also QGIS don't need the VRT, so there seem to be different ways to easily read the correct tiles at least?

Reply:
The copy method provides a new view (shallow copy) of the dataset is my understanding. So it shouldn't really matter in terms of memory.